### PR TITLE
completed data-structures-stacks  exercise - colin macrae

### PR DESCRIPTION
**Pull request by colin-macrae**

### CODE REVIEW

File: data-structures-stacks/count-values.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//     const top = stack.peek();
+//     popped.push(top);
+//     stack.pop();
+//   }
+//   return popped.length;
+// }
+
+export default function countValues(stack) {
+  let popped = 0;
+  for (let i = 0; i < 5; i++) {
```
Comment (RobertGardner) [inline data-structures-stacks/count-values.js:13]: Why only 5? You're lucky the tests don't use stacks with more than 5 elements in them. :-)

I think you saw that `while` is the best solution, which you now get to add to your list of things to learn more about. :-)

File: data-structures-stacks/max-value.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// I cheated on this one! 😏
+export default function countValues(stack) {
+  let max = -Infinity;
+  while (stack.peek() !== undefined) {
+    max = Math.max(max, stack.pop());
+  }
+  return max;
+}
+
+// export default function maxValue(stack) {
```
Comment (RobertGardner) [inline data-structures-stacks/max-value.js:10]: This solution down here works, too, but it makes a copy of the array so it's not an ideal solution.

**Review by RobertGardner — APPROVED** [review]
Nice!

File: data-structures-stacks/count-values.js (hunk 1 above)
Comment (colin-macrae) [reply data-structures-stacks/count-values.js:13]: Wel, because 5 worked 😂 

Ha, exactly.

File: data-structures-stacks/max-value.js (hunk 2 above)
Comment (colin-macrae) [reply data-structures-stacks/max-value.js:10]: Noted.

**Pull request merged**